smack-experimental/src/test/java/org/jivesoftware/smackx/file_metadata/FileMetadataElementTest.java

```diff
@@ -0,0 +1,137 @@
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.Date;
+
+import org.jivesoftware.smack.parsing.SmackParsingException;
+import org.jivesoftware.smack.test.util.SmackTestSuite;
+import org.jivesoftware.smack.test.util.TestUtils;
+import org.jivesoftware.smack.xml.XmlPullParserException;
+import org.jivesoftware.smackx.file_metadata.element.FileMetadataElement;
+import org.jivesoftware.smackx.file_metadata.element.child.DescElement;
+import org.jivesoftware.smackx.file_metadata.element.child.DimensionsElement;
+import org.jivesoftware.smackx.file_metadata.element.child.LengthElement;
+import org.jivesoftware.smackx.file_metadata.element.child.MediaTypeElement;
+import org.jivesoftware.smackx.file_metadata.element.child.NameElement;
+import org.jivesoftware.smackx.file_metadata.element.child.SizeElement;
+import org.jivesoftware.smackx.file_metadata.provider.FileMetadataElementProvider;
+import org.jivesoftware.smackx.hashes.HashManager;
+import org.jivesoftware.smackx.hashes.element.HashElement;
+
+import org.junit.jupiter.api.Test;
+import org.jxmpp.util.XmppDateTime;
+
+public class FileMetadataElementTest extends SmackTestSuite {
+
+    @Test
+    public void testSerialization() throws ParseException, XmlPullParserException, IOException, SmackParsingException {
+        Date date = XmppDateTime.parseDate("2015-07-26T21:46:00+01:00");
+        FileMetadataElement metadataElement = FileMetadataElement.builder()
+                .setModificationDate(date)
+                .setDimensions("1920x1080")
+                .addDescription("Picture of 24th XSF Summit")
+                .addDescription("Foto vom 24. XSF Summit", "de")
+                .addHash(new HashElement(HashManager.ALGORITHM.SHA_256, "2XarmwTlNxDAMkvymloX3S5+VbylNrJt/l5QyPa+YoU="))
+                .setLength(63000)
+                .setMediaType("text/plain")
+                .setName("text.txt")
+                .setSize(6144)
+                .build();
+
+        // Test serialization
+        final String expectedXml = "<file xmlns='urn:xmpp:file:metadata:0'>" +
+                "<date>2015-07-26T20:46:00.000+00:00</date>" +
+                "<dimensions>1920x1080</dimensions>" +
+                "<desc>Picture of 24th XSF Summit</desc>" +
+                "<desc xml:lang='de'>Foto vom 24. XSF Summit</desc>" +
+                "<hash xmlns='urn:xmpp:hashes:2' algo='sha-256'>2XarmwTlNxDAMkvymloX3S5+VbylNrJt/l5QyPa+YoU=</hash>" +
+                "<length>63000</length>" +
+                "<media-type>text/plain</media-type>" +
+                "<name>text.txt</name>" +
+                "<size>6144</size>" +
+                "</file>";
+        assertXmlSimilar(expectedXml, metadataElement.toXML().toString());
+
+        // Test deserialization
+        FileMetadataElement parsed = FileMetadataElementProvider.TEST_INSTANCE.parse(TestUtils.getParser(expectedXml));
```

Comment:
This should be split into two tests (serialization and deserialization) and use `@EnumSource(SmackTestUtil.XmlPullParserKind.class)` so that deserialization is performed with the two parsers available by Smack. See for example https://github.com/igniterealtime/Smack/blob/35a71f0131862b4cfdf3c4f5d31750245a3fc846/smack-extensions/src/test/java/org/jivesoftware/smackx/caps/provider/CapsExtensionProviderTest.java#L33

Reply:
Ah right, I was aware of the parametrization, but somehow writing tests the old way was too much of a routine for me :D